exporter/exporterhelper/persistent_queue.go

```diff
@@ -0,0 +1,343 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporterhelper
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/jaegertracing/jaeger/pkg/queue"
+	"go.uber.org/atomic"
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/extension/storage"
+)
+
+// persistentQueue holds the queue backed by file storage
+type persistentQueue struct {
+	logger     *zap.Logger
+	stopWG     sync.WaitGroup
+	produceMu  sync.Mutex
+	exitChan   chan int
```

Comment:
```suggestion
	exitChan   chan struct{}
```

It doesn't really send any `int`s does it?